api/src/main/java/org/apache/gravitino/file/FilesetCatalog.java

```diff
@@ -185,4 +185,6 @@ default String getFileLocation(NameIdentifier ident, String subPath, String loca
       throws NoSuchFilesetException, NoSuchLocationNameException {
     throw new UnsupportedOperationException("Not implemented");
   }
+
+  String listFiles(NameIdentifier ident);
```

Comment:
I assume that `listFiles` will only be available in the REST API. If so, is it necessary to add the `listFiles` method to the API module? 

Perhaps we can add an interface called `FilesetFileOps` in the core module, which includes the method `listFiles`. Otherwise, clients will see the `listFiles` method but won't be able to use it, causing confusion. WDYT? @jerryshao

Reply:
Do you think we need to have a client API for listFiles, not just REST API?

Reply:
I think only REST API is needed. If users want to use `listFiles` in the client, they should use GVFS.

Reply:
Ok, I see. Let's go with REST API first.